beam_nuggets/io/kafkaio.py

```diff
@@ -0,0 +1,139 @@
+        return (
+            pcoll 
+            | Create([self._attributes]) 
+            | ParDo(_ConsumeKafkaTopic(self._attributes))
+        )
+
+class _ConsumeKafkaTopic(DoFn):
+    """Internal ``DoFn`` to read from Kafka topic and return messages"""
+
+    def process(self, attributes):
+        consumer = KafkaConsumer(attributes["topic"], bootstrap_servers=attributes["servers"])
+            
+        for msg in consumer:
+            try:
+                yield (msg.key, msg.value.decode())
+            except Exception as e:
+                print(e)
+                continue
+
+class KafkaProduce(PTransform):
+    """A :class:`~apache_beam.transforms.ptransform.PTransform` for pushing messages
+    into an Apache Kafka topic. This class expects a tuple with the first element being the message key
+    and the second element being the message.
+
+    Examples:
+        Examples:
+        Pushing message to a Kafka Topic `notifications` ::
+
+            import apache_beam as beam
+            from apache_beam.options.pipeline_options import PipelineOptions
+            from beam_nuggets.io import kafkaio
+
+            with beam.Pipeline(options=PipelineOptions()) as p:
+                notifications = (p 
+                                 | "Creating data" >> beam.Create([('dev_1', '{"device": "0001", status": "healthy"}')])
+                                 | "Pushing messages to Kafka" >> kafkaio.KafkaProduce(
+                                                                                        topic='notifications',
+                                                                                        servers="localhost:9092"
+                                                                                    )
+                                )
+                notifications | 'Writing to stdout' >> beam.Map(print)
+
+        The output will be something like ::
+
+            ("dev_1", '{"device": "0001", status": "healthy"}')
+
+        Where the key is the Kafka topic published to and the element is the Kafka message produced
+    """
+
+    def __init__(self, topic=None, servers='127.0.0.1:9092'):
+        """Initializes ``KafkaProduce``
+
+        Args:
+            topic: Kafka topic to publish to
+            servers: list of Kafka servers to listen to
+
+        """
+        super(KafkaProduce, self).__init__()
+        self._attributes = dict(
+            topic=topic, 
+            servers=servers)
+
+    def expand(self, pcoll):
+        return (
+            pcoll 
+            | ParDo(_ProduceKafkaMessage(self._attributes))
+        )
+
+class _ProduceKafkaMessage(DoFn):
+    """Internal ``DoFn`` to publish message to Kafka topic"""
+
+    def __init__(self, attributes, *args, **kwargs):
+        super(_ProduceKafkaMessage, self).__init__(*args, **kwargs)
+        self.attributes = attributes
+
+    def process(self, element):
+        producer = KafkaProducer(bootstrap_servers=self.attributes["servers"])
```

Comment:
Since `DoFn.process(..)` is invoked for each element of the input `PCollection`, hence this will create as many new `KafkaProducer`:s. To have one `KafkaProducer` per `_ProduceKafkaMessage` we may achieve that using `DoFn.start_bundle(..)`; similar to [this](https://github.com/mohaseeb/beam-nuggets/blob/55441d9fd9b9e02ebe1c3434780472e1a7bf03d2/beam_nuggets/io/relational_db.py#L176).